### PR TITLE
Add SHTns recipe

### PR DESCRIPTION
**Pull request by peterspackman**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by peterspackman** [comment]
@conda-forge-admin, please ping team

File: recipes/shtns/bld.bat [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+) else (
+set GCC_ARCH=i686-w64-mingw32
+set EXTRA_FLAGS="-I/mingw-w64/%GCC_ARCH%/include/"
+)
+bash -lc "mkdir -p /tmp; ln -s ${LOCALAPPDATA}/Temp /tmp"
+bash -lc "echo $CYGWIN_PREFIX"
+bash -lc "export LDFLAGS=-L`cygpath -u $LIBRARY_LIB`; echo $LDFLAGS; ls -l `cygpath -u $LIBRARY_LIB`"
+bash -lc "cd src; export LDFLAGS=-L`cygpath -u $LIBRARY_LIB`; export PYTHON=`which python`; ./configure --disable-openmp --enable-python --build=$GCC_ARCH --host=$GCC_ARCH --target=$GCC_ARCH --prefix=$CYGWIN_PREFIX CFLAGS=$EXTRA_FLAGS"
+bash -lc "cd src; export CFLAGS=$EXTRA_FLAGS; make"
+bash -lc "cd src; make install"
```
Comment (scopatz) [inline recipes/shtns/bld.bat]: Can you make an install script for this.  Also each of these commands needs an exit condition check

File: recipes/shtns/build.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+SHTNS="shtns-3.3.1-r694"
+cd src
+./configure --enable-python --disable-openmp
```
Comment (scopatz) [inline recipes/shtns/build.sh]: this probably needs a `--prefix="${PREFIX}"`

File: recipes/shtns/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{% set name = "shtns" %}
+{% set version = "3.3.1" %}
+{% set revision = "r694" %}
```
Comment (scopatz) [inline recipes/shtns/meta.yaml]: We generally only build released versions of code

File: recipes/shtns/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    - m2-which  # [win]
+    - m2-tar  # [win]
+    - swig  # [win]
+    - {{ compiler('c') }}  # [unix]
+    - {{ compiler('m2w64_c') }}  # [win]
+  host:
+    - fftw
+    - m2w64-binutils  # [win]
+    - m2w64-gcc  # [win]
+    - numpy >=1.12
```
Comment (scopatz) [inline recipes/shtns/meta.yaml]: please see  https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy for numpy

File: recipes/shtns/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    - python
+  run:
+    - fftw
+    - {{ compiler('c') }}
```
Comment (scopatz) [inline recipes/shtns/meta.yaml]: Do you need a compiler at run time?

Comment (scopatz) [reply recipes/shtns/meta.yaml]: And this compiler seems inconsistent on windows

File: recipes/shtns/build.sh (hunk 2 above)
Comment (peterspackman) [reply recipes/shtns/build.sh]: you're correct, this should be explicit, fixed

File: recipes/shtns/bld.bat (hunk 1 above)
Comment (peterspackman) [reply recipes/shtns/bld.bat]: Thanks very much for taking a look, and the helpful comments!

As hard as I tried, getting an autotools build with conda working properaly on windows with the mixing of compilers/linkers was just not working. As such for now I've just made this not support windows. As far as I can find, there aren't even any build instructions for windows on the original author's repository/codebase so I think that's an acceptable outcome.

File: recipes/shtns/build.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+make
```
[the remark this answers is not in the document]
Comment (peterspackman) [reply recipes/shtns/build.sh]: I've updated this to use make install

File: recipes/shtns/meta.yaml (hunk 3 above)
Comment (peterspackman) [reply recipes/shtns/meta.yaml]: Perhaps the terminology is wrong (i.e. calling it a revision), but this is a released version of code: 3.3.1. It's just that the download has a suffix (in this case r694).

File: recipes/shtns/meta.yaml (hunk 4 above)
Comment (peterspackman) [reply recipes/shtns/meta.yaml]: I believe this is now fixed. Thanks for pointing this out.

File: recipes/shtns/meta.yaml (hunk 5 above)
Comment (peterspackman) [reply recipes/shtns/meta.yaml]: Re: the windows compiler, I've removed the windows specific build sections per the dropping of windows support.

Re: the compiler at runtime, I believe I might have originally put that in the requirements due to the dependency on libm (`-lm`), but that's now been removed.

File: recipes/shtns/meta.yaml (hunk 3 above)
Comment (scopatz) [reply recipes/shtns/meta.yaml]: OK

**Comment by scopatz** [comment]
Thanks @peterspackman!

**Pull request merged**